### PR TITLE
[Review] Correct unused parameters in column round and search

### PR DESCRIPTION
**Pull request by robertmaynard**
Starting in CUDA 11.3, nvcc will start to unconditionally warn about unused parameters on functions/methods that are in anonymous namespaces.

### CODE REVIEW

**Review by davidwendt — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]
LGTM, with a comment similar to https://github.com/rapidsai/cudf/pull/8390#pullrequestreview-671288046.

File: cpp/src/search/search.cu [hunk 1]
```diff
@@ -169,17 +169,17 @@ struct contains_scalar_dispatch {
 };
 
 template <>
-bool contains_scalar_dispatch::operator()<cudf::list_view>(column_view const& col,
-                                                           scalar const& value,
-                                                           rmm::cuda_stream_view stream)
+bool contains_scalar_dispatch::operator()<cudf::list_view>(column_view const&,
```
Comment (harrism) [inline cpp/src/search/search.cu:172]: Can we use `[[maybe_unused]]`?

```suggestion
bool contains_scalar_dispatch::operator()<cudf::list_view>([[maybe_unused]] column_view const&,
```

**Review by harrism — CHANGES_REQUESTED** [review]

File: cpp/src/search/search.cu (hunk 1 above)
Comment (robertmaynard) [reply cpp/src/search/search.cu:172]: Unfortunately not at this time: https://github.com/rapidsai/cudf/pull/8390#issuecomment-852083939

**Comment by robertmaynard** [comment]
rerun tests

**Review by mythrocks — APPROVED** [review]

**Comment by robertmaynard** [comment]
rerun tests

**Comment by mythrocks** [comment]
How odd. I'm not sure the test failures are related to this change.
```
opt/conda/envs/rapids/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:961: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:961: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:848: in exec_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
dask_cudf/__init__.py:6: in <module>
    from . import backends
dask_cudf/backends.py:124: in <module>
    ???
E   AttributeError: 'function' object has no attribute 'register'
```

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**